### PR TITLE
fix(components): [input] type hidden use class

### LINKED ISSUE #15864: [Component] [input] v-show使用初始化变量控制显示或隐藏时不起作用
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.19`
- Element Plus Version: `2.5.5`
- Browser / OS: `Chrome版本 121.0.6167.185（正式版本） （64 位）`
- Build Tool: `Webpack`

### Related Component

- `el-input`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgICA8ZWwtY2hlY2tib3ggdi1tb2RlbD1cImNoZWNrZWQxXCIgOmxhYmVsPVwiJ+WAvOS4uu+8micgKyBjaGVja2VkMVwiIHNpemU9XCJsYXJnZVwiIC8+XG4gICAgPGRpdj5cbiAgICAgICAgaW5wdXTkuI3nlJ/mlYjvvJogIDxlbC1pbnB1dCB2LXNob3c9XCJjaGVja2VkMVwiIHYtbW9kZWw9XCJtc2dcIj48L2VsLWlucHV0PlxuICAgICAgICBzd2l0Y2jlj6/ku6XnlJ/mlYjvvJogPGVsLXN3aXRjaCB2LXNob3c9XCJjaGVja2VkMVwiPjwvZWwtc3dpdGNoPlxuICAgIDwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgY2hlY2tlZDEgPSByZWYoZmFsc2UpXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG48L3NjcmlwdD5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

点击Playground链接可以直接查看效果

### What is Expected?

初始化时v-show的值为false，input控件应该是不显示的

### What is actually happening?

【初始化】时v-show的值为false，但是input控件还是显示出来了

### Additional comments

在示例中还放置了一个`el-switch`作为对比，是正常的，只有`el-input`有问题

### PR DESCRIPTION
**Pull request by chenxch**
Please make sure these boxes are checked before submitting your PR, thank you!

fixed #15864
- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by FrontEndDog — APPROVED** [review]

**Comment by btea** [comment]
The vue side seems to solve the related issues, I'm not sure if we need to adjust. https://github.com/vuejs/core/pull/10344

File: packages/theme-chalk/src/input.scss [hunk 1]
```diff
@@ -464,3 +464,8 @@
     }
   }
 }
+
+
+@include b(input-hidden) {
```
Comment (kooriookami) [inline packages/theme-chalk/src/input.scss:469]: What about `@include when(hidden)`

File: packages/components/input/src/input.vue [hunk 2]
```diff
@@ -223,6 +222,7 @@ const containerKls = computed(() => [
       slots.suffix || props.suffixIcon || props.clearable || props.showPassword,
     [nsInput.bm('suffix', 'password-clear')]:
       showClear.value && showPwdVisible.value,
+    [nsInput.b('hidden')]: props.type === 'hidden',
```
Comment (kooriookami) [inline packages/components/input/src/input.vue:225]: What about `.is('hidden')`

File: packages/theme-chalk/src/input.scss (hunk 1 above)
Comment (chenxch) [reply packages/theme-chalk/src/input.scss:469]: `hidden` is a type, not a state.

Comment (kooriookami) [reply packages/theme-chalk/src/input.scss:469]: OK~

**Review by kooriookami — APPROVED** [review]

File: packages/theme-chalk/src/input.scss [hunk 3]
```diff
@@ -464,3 +464,8 @@
...
+  display: none !important;
+}
```
Comment (btea) [inline packages/theme-chalk/src/input.scss:471]: Maybe we should add `appearance: none;`

**Review by btea — APPROVED** [review]

**Pull request merged**